app/src/minimal/kotlin/io/homeassistant/companion/android/location/LocationProvider.kt

```diff
@@ -0,0 +1,48 @@
+package io.homeassistant.companion.android.location
+
+import android.Manifest
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import android.os.CancellationSignal
+import androidx.annotation.RequiresPermission
+import androidx.core.content.ContextCompat
+import androidx.core.location.LocationManagerCompat
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import timber.log.Timber
+
+private val provider: String = if (SDK_INT >= Build.VERSION_CODES.S) {
+    LocationManager.FUSED_PROVIDER
+} else {
+    LocationManager.GPS_PROVIDER
+}
+
+@RequiresPermission(allOf = [Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION])
+suspend fun getLocation(context: Context): Location? {
+    val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as? LocationManager
+
+    if (locationManager == null) {
+        Timber.w("Fail to get LocationManager from context")
+        return null
+    }
+
+    val lastKnownLocation = locationManager.getLastKnownLocation(provider)
+    val now = System.currentTimeMillis()
+    // TODO maybe extract this into a constant since it also used in GeocodeSensorManager
+    return if (lastKnownLocation == null || now - lastKnownLocation.time > 300000) {
```

Comment:
Should this check be here? The geocoder sensor includes a check for recency, and this is inconsistent in behavior with the FLP function.

Reply:
I agree about inconsistency but since without this it won't ever update and we don't have any other sensors for the minimal flavor it's a tradeoff.
Where would it make sense to document this inconsistency?

Reply:
> without this it won't ever update

Would that also be true for someone using a device without Play Services? Any app that wants to get location will use the system API in that case. Keep in mind that the app's location tracking / Google Find my device network / Maps timeline frequently requests location if you have Play Services, making the 'last' location almost always recent.

> Where would it make sense to document this inconsistency?

A short comment inside the if block like `// Get current location because system APIs often do not have a recent enough location` would be enough in my opinion.

Reply:
> Would that also be true for someone using a device without Play Services? Any app that wants to get location will use the system API in that case. Keep in mind that the app's location tracking / Google Find my device network / Maps timeline frequently requests location if you have Play Services, making the 'last' location almost always recent.

Not sure to get your question. If GMS is not on the device and no app will request location then if we don't do what is in this function then the location won't ever update if I'm not wrong.

Reply:
My point was more that the lack of apps using the system APIs are probably the cause of no recent location. GMS APIs get a lot more location requests from system features keeping recent location updated.

Doesn't really matter at this point, seems fair to introduce it if needed to work well (but that function should only be for geocode).